src/views/product-tutorials-view/helpers/detect-and-reformat-learn-url.ts

```diff
@@ -0,0 +1,69 @@
+import { ProductOption, ThemeOption } from 'lib/learn-client/types'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getTutorial } from 'lib/learn-client/api/tutorial'
+import isAbsoluteUrl from 'lib/is-absolute-url'
+
+/**
+ * Given a URL string,  detect whether it is a collection or tutorial URL
+ * in the format used on learn.hashicorp.com.
+ *
+ * If it is, then reformat the URL to work with dev-dot's URL structure.
+ * Otherwise, return the URL unmodified.
+ */
+async function detectAndReformatLearnUrl(url: string): Promise<string> {
+  // Don't mess with absolute URLs
+  if (isAbsoluteUrl(url)) {
+    return url
+  }
+  // Gather info on whether this is a collection or tutorial URL
+  const urlObject = new URL(url, 'https://learn.hashicorp.com')
+  const pathParts = urlObject.pathname.split('/')
+  const isTutorialUrl =
+    pathParts[1] === 'tutorials' && isThemeOrProduct(pathParts[2])
```

Comment:
It could help to name what the pathPart is, `tutorialFilename` or `productSlug` etc. Since its difficult here to see how things are being combined.